### ISSUE TITLE
[Ch 01][Notebook][Major] TF-IDF 본문 수식에 smooth_idf +1 누락 — 같은 챕터 FAQ Q3 와 모순, 출력값도 재현 불가

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#85]
## 대상

- **챕터**: Ch 01 — 텍스트를 숫자로 (TF-IDF)
- **산출물**: Notebook (`01_tfidf/01_tfidf.ipynb` — `## 🛠️ 변형: TfidfVectorizer` 셀, id `cell017`)
- **기준 커밋**: `eed3dbe`
- **환경**: 정적 검토 + 실행본 `executed/01_tfidf.ipynb`(Colab, python 3.12.13) 대조 + 로컬 재현(sklearn 1.9.0 / datasets 5.0.1, 동일 `seed=42`). **Colab 전체 실행 완료** (2026-08-15 · CPU 런타임 · 63초 · 에러 0건)

## 실제 결과

본문의 TF-IDF 정의에 sklearn 기본값 `smooth_idf=True` 의 **`+1` 이 빠져 있습니다.**

```latex
\text{tfidf}(t, d) = \underbrace{\text{tf}(t, d)}_{\text{문서 } d \text{에서 } t \text{의 빈도}} \cdot \underbrace{\log\frac{1 + N}{1 + \text{df}(t)}}_{\text{희귀도 가중치 (IDF)}}
```

### 같은 챕터의 FAQ 와 자기모순입니다

FAQ Q3(`cell023`)는 `+1` 이 붙은 다른 식을 씁니다.

```latex
\text{idf}(t) = \log\frac{1 + N}{1 + \text{df}(t)} + 1
```

FAQ 쪽이 sklearn 실제 계산과 맞습니다. 한 챕터 안에서 같은 대상을 두 식으로 정의하고 있습니다.

### 어느 식으로도 바로 아래 출력이 재현되지 않습니다

`norm="l2"`(기본값) **행 정규화**가 챕터 어디에도 언급되지 않습니다. doc 0 의 `stalk`(tf=1, df=3, N=5000)로 검산하면:

| 계산 방식 | 값 |
|---|---|
| 본문 식 `tf·log((1+N)/(1+df))` | **7.1311** |
| FAQ 식 `tf·(log((1+N)/(1+df))+1)` | **8.1311** |
| FAQ 식 ÷ 행 L2 norm(33.6214) | **0.2418** |
| 노트북 실제 출력 (`cell019`) | **0.2418** |

학습자가 본문 식대로 손계산하면 바로 아래 표와 **약 30배** 차이가 납니다.

### idf 의 하한 설명도 틀렸습니다

본문 불릿:

> `idf`: 그 단어가 **얼마나 적은 문서에 등장했는가** (모든 문서에 흔할수록 **0에 가까워짐**)

`smooth_idf=True` 에서 idf 의 **최솟값은 1** 입니다(df=N → log 1 + 1 = 1). 같은 챕터 FAQ Q3 가 직접 "거기에 `+1` 이 더해져 IDF는 **1**" 이라고 적고 있어, 이 문장과도 어긋납니다.

## 기대 결과

본문 식을 sklearn 실제 계산에 맞추고 정규화를 한 줄 덧붙입니다.

```latex
\text{tfidf}(t, d) = \text{tf}(t, d) \cdot \left(\log\frac{1 + N}{1 + \text{df}(t)} + 1\right)
```

- 불릿 정정: "모든 문서에 흔할수록 **1에 가까워짐 — 이것이 최솟값**"
- 정규화 한 줄 추가 제안: "sklearn 은 마지막에 문서(행)별로 L2 정규화를 합니다. 그래서 아래 표의 값이 0~1 범위입니다 — 손으로 계산한 tf·idf 를 그 문서 벡터의 L2 norm 으로 나눈 값입니다."

## 재현 절차

노트북 재실행 불필요. 검산만으로 확인됩니다.

```python
import numpy as np
from datasets import load_dataset
from sklearn.feature_extraction.text import TfidfVectorizer
df = load_dataset("Yelp/yelp_review_full")["train"].shuffle(seed=42).select(range(5000)).to_pandas()
t  = TfidfVectorizer(max_features=10000);            X  = t.fit_transform(df["text"])
tn = TfidfVectorizer(max_features=10000, norm=None); Xn = tn.fit_transform(df["text"])
i = list(t.get_feature_names_out()).index("stalk")
raw = Xn[0].todense().A1
print(raw[i], np.linalg.norm(raw), raw[i] / np.linalg.norm(raw), X[0].todense().A1[i])
```

## 로그

```
8.131110 33.621442 0.241838 0.241838
```

본문 식으로 계산한 7.1311 로는 어떤 방식으로도 0.2418 이 나오지 않습니다.

## 전파 범위

세 산출물 모두에 있습니다 — 각각 직접 확인했습니다.

| 산출물 | 상태 |
|---|---|
| `01_tfidf/01_tfidf.ipynb` `cell017` | ❌ `+1` 없음, 정규화 미언급 |
| `pages/01-tfidf-variation.md:3` | ❌ `+1` 없음 (11행에 `+1` 판이 **중복**으로 함께 있음 → 별건) |
| `book/chapters/ch01_tfidf.tex:425` | ❌ `+1` 없음 → EPUB 영향 |

## 심각도

**Major** — 가이드 4쪽 Major 정의의 "잘못된 수식" 에 해당합니다. 실행은 정상이지만(`TfidfVectorizer` 가 내부에서 올바르게 계산), 학습자가 그대로 읽으면 TF-IDF 계산을 틀리게 배웁니다.

## 재실행 여부

마크다운 셀 수정이므로 **노트북 재실행은 불필요**합니다. WikiDocs 재변환 → EPUB 재빌드가 필요합니다.

## 담당

- 수정 담당자: 미정
- 재검수자: 등록자 (@modesta-jihye-lee)
- 연결: 챕터 검수 이슈 #25

**Comment by yoon-gu** [comment]
PASS — TF-IDF 본문 수식에 `smooth_idf` 의 `+1` 이 반영돼 같은 챕터 FAQ Q3 와의 모순이 해소됐습니다 (노트북 `cell017` · `pages/01-tfidf-variation.md` · `book/chapters/ch01_tfidf.tex` 3곳).

— PR #289 (master `6986584`, 2026-09-18). `origin/master` 기준으로 노트북·`executed/`·`pages/`·`tex` 전파까지 확인했습니다.
